xdsl/irdl/irdl.py

```diff
@@ -1483,6 +1483,12 @@ def wrong_field_exception(field_name: str) -> PyRDLOpDefinitionError:
                 if field_name in field_names:
                     # already registered value for field name
                     continue
+                if (
+                    field_name in annotations
+                    and field_name.isupper()
+                    and annotations[field_name].startswith("ClassVar[")
```

Comment:
This does not feel the cleanest, but I'm not sure what the proper way to do this would be...